packages/core/src/routes/applications/application.ts

```diff
@@ -129,6 +129,10 @@ export default function applicationRoutes<T extends AuthedRouter>(
     async (ctx, next) => {
       const { oidcClientMetadata, ...rest } = ctx.guard.body;
 
+      if (!EnvSet.values.isDevFeaturesEnabled && rest.type === ApplicationType.Protected) {
+        throw new Error('dev feature for protected app is not enabled');
+      }
```

Comment:
Should hide the new type from the input guard directly? This will take effect on the API docs as well.

Reply:
updated